src/store/mod.rs

```
@@ -419,6 +419,9 @@ impl Store {
     /// Adds a note tag to the list of tags that the client is interested in.
     pub fn add_note_tag(&mut self, tag: u64) -> Result<(), StoreError> {
         let mut tags = self.get_note_tags()?;
+        if tags.contains(&tag) {
+            return Err(StoreError::NoteTagAlreadyTracked(tag));
+        }
```

I wonder if we should return an error here or use the same approach as [BTreeSet::insert()](https://doc.rust-lang.org/std/collections/struct.BTreeSet.html#method.insert) uses - e.g., return `true` if the value was present in the set of tags, and false otherwise.

More concretely, we'd change the return type of this function to `Result<bool, StoreError>` and then this statement would become:
```Rust
if tags.contains(&tag) {
    return Ok(true)
}
```

Main motivation is that adding duplicate tag is not really an error - we just shouldn't add it (though, we may want to show a warning about this at some point later).

---

I think it's a good approach, I think we should use the same `true`/`false` return conditions as  [BTreeSet::insert()](https://doc.rust-lang.org/std/collections/struct.BTreeSet.html#method.insert):

> Returns whether the value was newly inserted. That is:
> 
> If the set did not previously contain an equal value, `true` is returned.
> If the set already contained an equal value, `false` is returned, and the entry is not updated.